src/test/java/JsonUtilityTest.java

```diff
@@ -12,94 +12,138 @@
+    printWriter.close();
   }
 
   @Test
   public void sendStringToResponse() throws IOException {
-    // String should be sent successfully and in the expected format
+    // Sends an object of type String to response as json.
+    // Response should have the same contents as String object.
     JsonUtility.sendJson(response, STRING_OBJECT);
+    printWriter.flush();
 
-    String expected = gson.toJson(STRING_OBJECT);
-    String actual = stringWriter.toString();
+    String actual = gson.fromJson(stringWriter.toString(), String.class);
 
-    Assert.assertTrue(actual.contains(expected));
+    Assert.assertEquals(STRING_OBJECT, actual);
   }
 
   @Test
   public void sendListToResponse() throws IOException {
-    // List should be sent successfully and in the expected format
+    // Sends an object of type List to response as json.
+    // Response should have the same contents as List object.
     JsonUtility.sendJson(response, LIST_OBJECT);
+    printWriter.flush();
 
-    String expected = gson.toJson(LIST_OBJECT);
-    String actual = stringWriter.toString();
+    String actualString = stringWriter.toString();
+    Type type = new TypeToken<List<Integer>>() {}.getType();
+    List<Integer> actual = gson.fromJson(actualString, type);
 
-    Assert.assertTrue(actual.contains(expected));
+    Assert.assertEquals(LIST_OBJECT, actual);
   }
 
   @Test
   public void sendMapToResponse() throws IOException {
-    // Map should be sent successfully and in the expected format
+    // Sends an object of type Map to response as json.
+    // Response should have the same contents as Map object.
     JsonUtility.sendJson(response, MAP_OBJECT);
+    printWriter.flush();
+
+    String actualString = stringWriter.toString();
+    Type type = new TypeToken<Map<String, Integer>>() {}.getType();
+    Map<String, Integer> actual = gson.fromJson(actualString, type);
+
+    Assert.assertEquals(MAP_OBJECT.get("One"), actual.get("One"));
+    Assert.assertEquals(MAP_OBJECT.get("Two"), actual.get("Two"));
+    Assert.assertEquals(MAP_OBJECT.get("Three"), actual.get("Three"));
+  }
+
+  @Test
+  public void sendNullToResponse() throws IOException {
+    // Sends null to response as json.
+    // Response should contain only a newline character as a result of println.
+    JsonUtility.sendJson(response, null);
+    printWriter.flush();
+
+    Assert.assertEquals("\n", stringWriter.toString());
+  }
+
+  @Test
+  public void sendCustomClassToResponse() throws IOException {
+    // Sends an object of type CustomClass to response as json.
+    // Response should have the same contents as CustomClass object.
+    JsonUtility.sendJson(response, CUSTOM_OBJECT);
+    printWriter.flush();
 
-    String expected = gson.toJson(MAP_OBJECT);
-    String actual = stringWriter.toString();
+    String actualString = stringWriter.toString();
+    Type type = new TypeToken<Map<String, String>>() {}.getType();
+    Map<String, String> actual = gson.fromJson(actualString, type);
```

Comment:
Can you deserialize to the target class:
```
CustomClass actual = gson.fromJson(actualString, CustomClass.class;
```

Reply:
@antmar It seems like I would have to write my own deserializer and then register `CustomClass` as a type adapter in order to run that line of code. Would you recommend this to be done?

Reply:
Perhaps I misunderstood the documentation, but it seems like you don't need a custom deserializer: https://github.com/google/gson/blob/master/UserGuide.md#TOC-Object-Examples

Please don't waste too much time on this, however.

Reply:
I found the issue by reading the example - it was because CustomClass didn't have a no-args constructor so I simplified CustomClass and deserialized CustomClass to the target class. Thanks for the link Anton!